agent/api/ecsclient/client.go

```diff
@@ -191,11 +201,36 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 
 	resp, err := client.standardClient.RegisterContainerInstance(&registerRequest)
 	if err != nil {
-		log.Error("Could not register", "err", err)
+		log.Error("Could not register", "error", err)
 		return "", err
 	}
 	log.Info("Registered!")
-	return *resp.ContainerInstance.ContainerInstanceArn, nil
+	err = validateRegisteredAttributes(registerRequest.Attributes, resp.ContainerInstance.Attributes)
+	return *resp.ContainerInstance.ContainerInstanceArn, err
+}
+
+func attributesToMap(attributes []*ecs.Attribute) map[string]string {
+	attributeMap := make(map[string]string)
+	attribs := attributes
+	for i := range attribs {
+		attributeMap[aws.StringValue(attribs[i].Name)] = aws.StringValue(attribs[i].Value)
+	}
+	return attributeMap
+}
+
+func validateRegisteredAttributes(expectedAttributes, actualAttributes []*ecs.Attribute) error {
+	var err error
+	expectedAttributesMap := attributesToMap(expectedAttributes)
+	actualAttributesMap := attributesToMap(actualAttributes)
+	for key, val := range expectedAttributesMap {
+		if actualAttributesMap[key] != val {
+			log.Error("Error registering an attribute", "attribute", key)
+			err = utils.NewAttributeError("Attribute validation failed")
```

Comment:
I had a question regarding this method in the previous version of the file, which is still not addressed:

> Can you just return error here? Or, is the intent to print all invalid attributes? It might be better to refactor this method into a method that returns a slice of of invalid attributes and an error to make that clear and log invalid attributes where this gets invoked if that's the case.